src/config/ConfigBuilder.ts

```diff
@@ -223,10 +224,76 @@ export class ConfigBuilder {
 
     private handleProperties(jsonObject: any): void {
         for (const key in jsonObject) {
-            if (!PROPERTY_SET.has(key)) {
-                throw new RangeError(`Unexpected property '${key}' is passed to the Hazelcast Client`);
+            let value = jsonObject[key];
+            try {
+                switch (key) {
+                    case 'hazelcast.client.heartbeat.interval':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.client.heartbeat.timeout':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.client.invocation.retry.pause.millis':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.client.invocation.timeout.millis':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.client.internal.clean.resources.millis':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.client.cloud.url':
+                        value = tryGetString(value);
+                        break;
+                    case 'hazelcast.client.statistics.enabled':
+                        value = tryGetBoolean(value);
+                        break;
+                    case 'hazelcast.client.statistics.period.seconds':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.invalidation.reconciliation.interval.seconds':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.invalidation.max.tolerated.miss.count':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.invalidation.min.reconciliation.interval.seconds':
+                        value = tryGetNumber(value);
+                        break;
+                    case 'hazelcast.logging.level':
+                        tryGetEnum(LogLevel, value); // only validate
```

Comment:
can't we change the implementation so that we use the value? I see that in the LoggingService, we already do that. Am I missing something?

Reply:
we need to preserve value as string. If we change it to be number via tryGetEnum, the logging service will not work. Should I update it too?

Reply:
Implemented in ef07339c